classes/01class/exercises/c01-git03/jayanath/README.md

```diff
@@ -0,0 +1,35 @@
+# c01-git03
+
+## Questions (answer inline)
+
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+```
+Yes. The repository size increased from 100K to 20M. 
+```
+
+2. What is the reason for this problem?
+```
+Git keeps every version of each and every file and that design does not work well with large files.
+It significantly increases the size of the repository with each new version.
+```
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+```
+git filter-branch --tree-filter 'rm -rf path/to/the/bigfile' HEAD
+git push
+
+```
+
+4. What are the consequences of the previous action?
+```
+"git filter-branch has a plethora of pitfalls that can produce non-obvious manglings" 
+- source: git documentation [https://git-scm.com/docs/git-filter-branch]
+
+I tried to use git-filter-repo instead [https://github.com/newren/git-filter-repo/]
```

Comment:
Alright, I see your point.

In fact, when using trunk-based development approach in git, just fixing `master` would be enough as people need to synchronize with master before releasing in product. 

So a filter-branch or rebase in master would solve the problem and every other dev should fix their dev branches.